app/models/template.rb

```diff
@@ -5,10 +5,54 @@ class Template < ApplicationRecord
 
   validates :title, :presence => :title
 
+  before_destroy :delete_passwords
+
   def self.seed
     template = find_or_create_by!(:title => 'Basic')
     template.update_attributes(
-      :description => 'A basic approval workflow that supports multi-level approver groups through email notification'
+      :description     => 'A basic approval workflow that supports multi-level approver groups through email notification',
+      :process_setting => seed_process_setting(template.process_setting),
+      :signal_setting  => seed_signal_setting(template.signal_setting)
     )
   end
+
+  private_class_method def self.seed_process_setting(old_setting)
+    return nil unless ENV['KIE_SERVER_HOST']
+    seed_bpm_setting(old_setting).merge('process_id' => ENV['BPM_BML_PROCESS_ID'])
+  end
+
+  private_class_method def self.seed_signal_setting(old_setting)
+    return nil unless ENV['KIE_SERVER_HOST']
+    seed_bpm_setting(old_setting).merge('signal_name' => ENV['BPM_BML_SIGNAL_NAME'])
+  end
+
+  private_class_method def self.seed_bpm_setting(old_setting)
+    old_password_id = old_setting.try(:[], 'password')
+    old_password = Encryption.find(old_password_id) if old_password_id
+
+    new_password_id =
+      if old_password.try(:secret) == ENV['KIE_SERVER_PASSWORD']
+        old_password_id
+      else
+        old_password.try(:delete)
+        Encryption.create!(:secret => ENV['KIE_SERVER_PASSWORD']).id
+      end
+
+    {
+      'container_id' => ENV['KIE_CONTAINER_ID'],
+      'password'     => new_password_id,
+      'username'     => ENV['KIE_SERVER_USERNAME'],
+      'host'         => ENV['KIE_SERVER_HOST']
+    }
+  end
+
+  private
+
+  def delete_passwords
+    process_password_id = process_setting.try(:[], 'password')
+    Encryption.delete(process_password_id) if process_password_id
```

Comment:
Why not use `.destroy!` here as opposed to `.delete`?

Reply:
No difference. There is no direct association between `Encryption` and any other models

Reply:
But, if you call `.destroy!` and there is an Exception it will cancel the call back and the error will be bubbled up.  With `.delete` you will not get any feedback and the callback will continue as if nothing was wrong.